examples/high_level_plots.py

```diff
@@ -0,0 +1,90 @@
+"""Produce roc curves from tagger output and labels."""
+# from pathlib import Path
+
+# import h5py
+import numpy as np
+
+from puma.hlplots.results import Results
+from puma.hlplots.tagger import Tagger
+from puma.utils import get_dummy_2_taggers, logger
+
+# The line below generates dummy data which is similar to a NN output
+df = get_dummy_2_taggers(add_pt=True)
+class_ids = [0, 4, 5]
+# Remove all jets which are not trained on
+df.query(f"HadronConeExclTruthLabelID in {class_ids}", inplace=True)
+df.query("pt < 250e3", inplace=True)
+
+logger.info("Start plotting")
+
+tagger_args = {
+    "perf_var": df["pt"] / 1e3,
+    "is_light": df["HadronConeExclTruthLabelID"] == 0,
+    "is_c": df["HadronConeExclTruthLabelID"] == 4,
+    "is_b": df["HadronConeExclTruthLabelID"] == 5,
+}
+
+
+dips = Tagger("dips", template=tagger_args)
+dips.label = "dummy DIPS ($f_{c}=0.005$)"
+dips.f_c = 0.005
+dips.f_b = 0.04
+dips.colour = "#AA3377"
+dips.extract_tagger_scores(df)
```

Comment:
Maybe I am missing something, but looking at the example here, it looks to me like one could (accidentally) define `is_light`, `is_c`, ... masks which do not match the dataframe given in `tagger.exact_tagger_scores(df)`.

So in theory, if you have two different dataframes in one script, one could mess this up I think.
Maybe we could hide the mask calculation inside the class and just define a dict with label, value pairs. E.g.
```py
tagger_args = {
    "is_light": {"HadronConeExclTruthLabelID": 0},
    "is_c": {"HadronConeExclTruthLabelID": 4},
    "is_b": {"HadronConeExclTruthLabelID": 5},
}
```

We could then also allow lists in the values (e.g. `"bbjets": {"HadronConeExtendedTruthLabelID" : [55, 5]}` and use `np.isin()` under the hood.

Reply:
I would keep it like it is implemented now though I can change the example such that it uses 2 data frames where it should be clear then that one needs to change it